skiko/src/awtMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.kt

```diff
@@ -86,6 +84,11 @@ internal class MetalRedrawer(
         onContextInit()
     }
 
+    fun drawSync() {
```

Comment:
Rewired logic to make redrawImmediately wait for barriers assigned to FrameDispatcher.

Reply:
`redrawImmediately` is supposed to draw the frame as fast as possible, not waiting for the next vsync. So, when there is a vsync, we already have the new content.

The question was - is `drawSync` behaves the same as the old `redrawImmediately`. I tested, and it seems not, as we wait for vsync before committing the frame. So, if we still need it for tests, let's return it, and revert `redrawImmediately`.

Reply:
Then we need to figure out what this method does exactly. Next vsync is almost as fast as possible. The fastest possible way is to schedule the frame immediately and pray, that there wasn't a frame scheduled for current vsync already, otherwise newest one will be rendered, but its drawable will be discarded. What do we actually want here?

`immediately` works nice in drawable-less context (like off-screen rendering), but rendering on window is tightly coupled with vsync.